src/config/_colors.scss

```diff
@@ -4,7 +4,47 @@ $colors: map-get($variables, 'colors');
 $statuses: map-get($variables, 'colorStatuses');
 $combinations: map-get($variables, 'colorCombinations');
 
-@mixin combination($selector, $combination, $colors, $smoothing: false) {
+// Variables
+$primary-navy: map-get($colors, 'primary-navy');
+$primary-navy-70t: map-get($colors, 'primary-navy-70t');
+$primary-navy-50t: map-get($colors, 'primary-navy-50t');
+$primary-navy-30t: map-get($colors, 'primary-navy-30t');
+
+$primary-blue: map-get($colors, 'primary-blue');
+$primary-blue-70t: map-get($colors, 'primary-blue-70t');
+$primary-blue-50t: map-get($colors, 'primary-blue-50t');
+$primary-blue-30t: map-get($colors, 'primary-blue-30t');
+
+$primary-red: map-get($colors, 'primary-red');
+$primary-red-70t: map-get($colors, 'primary-red-70t');
+$primary-red-50t: map-get($colors, 'primary-red-50t');
+$primary-red-30t: map-get($colors, 'primary-red-30t');
+
+$secondary-grey: map-get($colors, 'secondary-grey');
+$secondary-grey-70t: map-get($colors, 'secondary-grey-70t');
+$secondary-grey-50t: map-get($colors, 'secondary-grey-50t');
+$secondary-grey-30t: map-get($colors, 'secondary-grey-30t');
+
+$secondary-white: map-get($colors, 'secondary-white');
+$secondary-white-70t: map-get($colors, 'secondary-white-70t');
+$secondary-white-50t: map-get($colors, 'secondary-white-50t');
+$secondary-white-30t: map-get($colors, 'secondary-white-30t');
+
+$secondary-orange: map-get($colors, 'secondary-orange');
+$secondary-orange-70t: map-get($colors, 'secondary-orange-70t');
+$secondary-orange-50t: map-get($colors, 'secondary-orange-50t');
+$secondary-orange-30t: map-get($colors, 'secondary-orange-30t');
+
+$secondary-pink: map-get($colors, 'secondary-pink');
+$secondary-pink-70t: map-get($colors, 'secondary-pink-70t');
+$secondary-pink-50t: map-get($colors, 'secondary-pink-50t');
+$secondary-pink-30t: map-get($colors, 'secondary-pink-30t');
+
+$base-black: map-get($colors, 'base-black');
+$base-white: map-get($colors, 'base-white');
+
+// Mixins
+@mixin combination($selector, $combination, $b, $smoothing: false) {
```

Comment:
@kimpenguin @KISS wanted to share the method for defining accessible color palettes. The base reason for all of the "inherit" comments I'm making (make this color inherit) in this PR is because this mixin provides an easy way to control color combinations through a single parent selector. So, all color definitions in patterns can be moved into the variable that feeds this mixin. In this way, we ensure that certain color combinations are accessible. In the `config/variables.js` file there is an object called [colorCombinations](https://github.com/CityOfNewYork/nyco-patterns/blob/c8e7df4ed8030d9c3867c8f1b0505b25aa1e51a2/config/variables.js#L98) where they are defined.

So, for example, the color combination for the `.color-light-background` (which is basically white backgrounds) class is this;

```
    'light-background': {
      'color': 'base-black',
      'headings': 'primary-navy',
      'hyperlinks': 'primary-blue',
      'visited': 'primary-blue',
      'hover': 'primary-navy',
      'background-color': 'base-white'
    }
```
This will control the text color, heading colors, hyperlinks, and background color. If we instruct other users to stick to these combinations, we can insure accessibility standards are utilized, meaning we'll know they are WCAG AA contrast compliant.

You can see in the [mixin](https://github.com/CityOfNewYork/nyco-patterns/blob/c8e7df4ed8030d9c3867c8f1b0505b25aa1e51a2/src/config/_colors.scss#L26) where this comment is placed how that variable is used to style child elements of the selector. This could be extended to include input definitions as well;

```
    'light-background': {
      'color': 'base-black',
      'headings': 'primary-navy',
      'hyperlinks': 'primary-blue',
      'visited': 'primary-blue',
      'hover': 'primary-navy',
      'background-color': 'base-white',
      // text inputs, selects, and textarea etc.
      'input-border': 'secondary-grey',
      'input-color': 'base-black',
      'input-placeholder': 'secondary-grey',
      // checkboxes
      'checkbox-border': 'primary-blue',
      ... etc. ...
    }
```

If an element has the color property `inherit` set to it, then it will inherit the parent color definition defined by the combination. This is why it is preferred to never define colors in the patterns at the "atomic" or element level. There will be exceptions for components and objects. Also, individual colors can always be overridden by the text, border, and background color utilities;

- https://tailwindcss.com/docs/text-color
- https://tailwindcss.com/docs/background-color
- https://tailwindcss.com/docs/border-color

ACCESS has not been completely set up to work this way for inputs, checkboxes, etc. Another reason it's not a great example to pull configuration and base styles from. Some combinations have been set up but it's a work in progress. Really sorry I didn't explain this before.

Reply:
Oh, obviously, there's other magic we can do with color contrast functions and SASS if you would like to explore that https://medium.com/dev-channel/using-sass-to-automatically-pick-text-colors-4ba7645d2796

Reply:
I'm not fully clear on this. Can we go over it during the pattern meeting next week? 

For now, do you want to have the color variables removed and references to them changed to `inherit`? And then any styles we want to implement should get defined in their respective mixin (in your example: `border-color` would be set in `light-background: {}`) and then we add the class to the html markup?

Reply:
_For now, do you want to have the color variables removed and references to them changed to inherit?_ Yes, that would be preferred.

_And then any styles we want to implement should get defined in their respective mixin (in your example: border-color would be set in light-background: {}) and then we add the class to the html markup?_ When you add color defs to the colorCombinations configuration, and then update the color combinations mixin in /src/config/_colors.scss, you won't need to add any color utilities to the markup.

Happy to go over in person!